src/promptflow/promptflow/_sdk/_submitter/test_submitter.py

```diff
@@ -448,21 +496,91 @@ def exec_with_inputs(self, inputs):
             connections=connections,
             storage=storage,
         )
 
         try:
             # validate inputs
             flow_inputs, _ = self.resolve_data(inputs=inputs, dataplane_flow=self.dataplane_flow)
-            line_result = async_run_allowing_running_loop(flow_executor.exec_line_async, inputs, index=0)
-            # line_result = flow_executor.exec_line(inputs, index=0)
-            if isinstance(line_result.output, dict):
+            line_result = async_run_allowing_running_loop(flow_executor.exec_line_async, inputs, index=0,
+                                                          allow_generator_output=allow_generator_output)
+            if not isinstance(line_result, AsyncGeneratorType) and isinstance(line_result.output, dict):
                 # Remove line_number from output
                 line_result.output.pop(LINE_NUMBER_KEY, None)
             return line_result
         finally:
             flow_executor.destroy()
+
+    def _chat_flow(self, inputs, chat_history_name, environment_variables: dict = None, show_step_output=False):
+        """
+        Interact with Chat Flow. Do the following:
+            1. Combine chat_history and user input as the input for each round of the chat flow.
+            2. Each round of chat is executed once flow test.
+            3. Prefix the output for distinction.
+        """
+
+        @contextlib.contextmanager
+        def change_logger_level(level):
+            origin_level = logger.level
+            logger.setLevel(level)
+            yield
+            logger.setLevel(origin_level)
+
+        init(autoreset=True)
+        chat_history = []
+        # generator_record = {}
+        input_name = next(
+            filter(lambda key: self.dataplane_flow.inputs[key].is_chat_input, self.dataplane_flow.inputs.keys())
+        )
+        output_name = next(
+            filter(
+                lambda key: self.dataplane_flow.outputs[key].is_chat_output,
+                self.dataplane_flow.outputs.keys(),
+            )
+        )
+
+        # Pass connections to avoid duplicate calculation (especially http call)
+        connections = SubmitterHelper.resolve_used_connections(
+            flow=self.flow,
+            tools_meta=CSharpExecutorProxy.get_tool_metadata(
+                flow_file=self.flow.flow_dag_path,
+                working_dir=self.flow.code,
+            ),
+            client=self._client,
+        )
+
+        while True:
+            try:
+                print(f"{Fore.GREEN}User: ", end="")
+                input_value = input()
+                if not input_value.strip():
+                    continue
+            except (KeyboardInterrupt, EOFError):
+                print("Terminate the chat.")
+                break
+            inputs = inputs or {}
+            inputs[input_name] = input_value
+            inputs[chat_history_name] = chat_history
+            with change_logger_level(level=logging.WARNING):
```

Comment:
why do we need to specially change level here?

Reply:
just copy from python chat_flow

Reply:
I mean we might be able to reduce some dulplicate here. But we can do it in future.